dapp/src/components/News.js

```diff
@@ -0,0 +1,191 @@
+import { Card, Select } from '@originprotocol/origin-storybook'
+import withIsMobile from 'hoc/withIsMobile'
+import Image from 'next/image'
+import React, { useEffect, useState } from 'react'
+import { getStrapiMedia } from '../../lib/media'
+import { assetRootPath } from 'utils/image'
+
+const Category = ({ categories, setCategory }) => {
+  const [open, setOpen] = useState(false)
+  //const categories = ['All news', 'News', 'Food', 'Nature', 'Tech', 'Story']
+  const capitalize = (name) => {
+    return name.slice(0, 1).toUpperCase() + name.slice(1, name.length)
+  }
+
+  const categoriesFormatted = [
+    {
+      id: null,
+      name: 'All articles',
+      unavailable: false,
+    },
+  ].concat(
+    categories.map((category) => {
+      return {
+        id: category.slug,
+        name: capitalize(category.name),
+        unavailable: false,
+      }
+    })
+  )
+
+  return (
+    <div className="pl-0 w-96 pt-4 text-black">
+      <Select
+        options={categoriesFormatted}
+        onSelect={(value) => {
+          setCategory(value.id)
+        }}
+      />
+    </div>
+  )
+}
+
+const News = ({ isMobile, articles, meta, categories }) => {
+  const [loaded, setLoaded] = useState(false)
+
+  useEffect(() => {
+    setLoaded(true)
+  }, [])
+
+  const [category, setCategory] = useState('')
+  const [page, setPage] = useState(1)
+  const [pageNumbers, setPageNumbers] = useState([])
+
+  const articlePages = Math.ceil(
+    (category
+      ? articles.filter((article) => article.slug === category.slug).length
```

Comment:
Don't this would work in all cases. Say, I have 200 articles and 10 categories (each with 20 articles). If the first request only returns the first 100 articles, some articles under categories would be missing and there would be no pagination that'd work either

Reply:
Perhaps we could try using params in route to render these pages. Like `/category` would render for all categories, whereas `/category/defi` would load the data for that specific category. Better to trust the server on pagination and also if we pregenerate all category pages during build, it won't even affect the performance